src/test/groovy/graphql/NonNullHandlingTest.groovy

```diff
@@ -273,5 +276,124 @@ class NonNullHandlingTest extends Specification {
 
     }
 
+    def "#561 - null entry in non null list type with non null wrapper list"() {
 
+        given:
+
+
+        GraphQLOutputType parentType = newObject()
+                .name("parentType")
+                .field(newFieldDefinition().name("nonNullListWithNull")
+                .type(nonNull(list(nonNull(GraphQLString)))))
+                .build()
+
+        GraphQLOutputType topType = newObject()
+                .name("topType")
+                .field(newFieldDefinition().name("nullParent")
+                .type(nonNull(parentType)))
+                .field(newFieldDefinition().name("nonNullParent")
+                .type(nonNull(parentType)))
+                .build()
+
+        GraphQLSchema schema = newSchema().query(
+                newObject()
+                        .name("RootQueryType")
+                        .field(
+                        newFieldDefinition()
+                                .name("top")
+                                .type(nonNull(topType)) // non nullable grand parent
+                                .dataFetcher({ env -> new ContainingObject() })
+
+                ))
+                .build()
+
+        def query = """
+        query { 
+            top {
+                nonNullParent {
+                    nonNullListWithNull
+                }
+            }
+        }
+        """
+
+        def result = GraphQL
+                .newGraphQL(schema)
+                .queryExecutionStrategy(executionStrategy)
+                .build()
+                .execute(executionInput(query))
+
+        expect:
+
+        result != null
+        result.errors.size() == 1
+        result.data == null
+
+        where:
+
+        strategyName | executionStrategy
+        'executor'   | new ExecutorServiceExecutionStrategy(commonPool())
+        'simple'     | new AsyncExecutionStrategy()
```

Comment:
I would not call it `simple` anymore, because we don't have a `simple` strategy anymore.